scripts/creation-portal-migration/orgDetails.js

```diff
@@ -0,0 +1,129 @@
+
+/**
+* To get the userdata from creation portal
+* @method
+* @name getUserIds
+**/
+const getUserIds = async () => {
+  try {
+    const db = await createDBInstance();
+    const data = await findAll(CONFIG.DB.TABLES.solutions, {
+      programId: { $exists: true },
+      type: { $in: ["observation", "survey"] },
+    });
+
+    const userIds = data.map((solution) => solution.author);
+    const solutions = data.map((solution) => {
+      const programName = `${solution.name} sourcing project`
+      return {
+        userId: solution.author,
+        solutionId: solution._id,
+        solutionName: solution.name,
+        programId: solution?.migrationReference?.sourcingProgramId || "",
+        programName: programName,
+      };
+    });
+    let uniqUsers = uniq(userIds);
+    uniqUsers = compact(uniqUsers);
+    const usersList = await searchUser(uniqUsers);
+    const openSaberOrg = await getOpenSaberUserOrgId();
+    const d = writeToCSVFile(usersList, uniqUsers, openSaberOrg, solutions);
+    console.log(`\n migratedCount userIds`, d);
+  } catch (err) {
+    console.log(`Error while migrating : ${err}`);
+    throw new Error("Error occured", err);
+  }
+};
+
+/**
+* Write the data to csv file
+* @method
+* @name writeToCSVFile
+* @param {Object[]} users - users
+* @param {String[]} usersIdsInDb - usersIdsInDb
+* @param {Object[]} openSaberOrg - 
+"User_Org": [
+  {
+      "osUpdatedAt": "2021-05-11T06:07:45.534Z",
+      "osCreatedAt": "2021-05-11T06:07:45.534Z",
+      "@type": "User_Org",
+      "roles": [
+          "user",
+          "sourcing_reviewer"
+        ],
+      "osid": "0933fa92-729f-4a77-b5d7-cba40f68b4eb",
+      "userId": "d59873a0-40ea-461b-9402-ab090932f92d",
+      "orgId": "7c5a96ca-bef8-4027-8736-4fa1ae6f9180"
+    }
+  ]
+* @param {Object[]} solutions - solutions
+**/
+writeToCSVFile = (users, usersIdsInDb, openSaberOrg, solutions) => {
+  const filename = __dirname + "/SL-DataMapping.csv";
+  fs.writeFile(
+    filename,
+    extractAsCSV(users, usersIdsInDb, openSaberOrg, solutions),
+    (err) => {
+      if (err) {
+        console.log("Error writing to csv file", err);
+      } else {
+        console.log(`saved as ${filename}`);
+        process.exit();
+      }
+    }
+  );
+};
+
+/**
+* Extract the data in csv format
+* @method
+* @name extractAsCSV
+* @param {Object[]} users - users
+* @param {String[]} usersIdsInDb - usersIdsInDb
+* @param {Object[]} openSaberOrg - 
+"User_Org": [
+  {
+      "osUpdatedAt": "2021-05-11T06:07:45.534Z",
+      "osCreatedAt": "2021-05-11T06:07:45.534Z",
+      "@type": "User_Org",
+      "roles": [
+          "user",
+          "sourcing_reviewer"
+        ],
+      "osid": "0933fa92-729f-4a77-b5d7-cba40f68b4eb",
+      "userId": "d59873a0-40ea-461b-9402-ab090932f92d",
+      "orgId": "7c5a96ca-bef8-4027-8736-4fa1ae6f9180"
+    }
+  ]
+* @param {Object[]} solutions - solutions
+**/
+const extractAsCSV = (users, usersIdsInDb, openSaberOrg, solutions) => {
```

Comment:
@swetha-aggidevara users data will have userId and userName right? As per our discussion with piyush hw said
**Use userId and userName as srcOrgAdminId and srcOrgAdminUserName.
Use userId and userName as contributorOrgAdminId,contributorOrgAdminUserName.**
So can we make changes to write these values to csv also?

Reply:
@VISHNUDAS-tunerlabs  This data can't be is not from the API'S used in the script,
It has to be done manually because we are only finding the user data in the api not admindata.

so we can't update the csv

Reply:
@swetha-aggidevara remember piyush confirmed that userId and userName we find from user data is used as srcOrgAdminId, srcOrgAdminUserName, contributorOrgAdminId, contributorOrgAdminUserName values? if it is we can do this no? anyway we are finding user data

Reply:
@VISHNUDAS-tunerlabs 
**To generate the csv from the script**
- We have to get userid from mongo
- Search for those userids using the API,  this will return only userdata present in Mongo.
- The filter user  in the above search api is to search only by userids.
- If user is present then the csv will be updated with userId and rootOrgId, else user has to be created and update the user data in the sheet manually.
- but to get srcOrgadmin and contributorOrgadmin we have to search the users using the filter : rootOrgId of the user present in Mongo which should be done manually to update the sheet.

**Please go through the document once again**